### ISSUE TITLE
[ServiceBus] time_to_live property triggering OverflowError: mktime argument out of range in core regression test

### ISSUE THREAD

**Issue by yunhaoling** [Azure/azure-sdk-for-python#19297]
- **Package Name**: azure-servicebus
- **Package Version**: 7.3.0
- **Operating System**: Linux
- **Python Version**: 3.7.10

**Describe the bug**
in the internal pipeline `python - core` 20210616.1, test `test_servicebus_message_repr_with_props` consistently fails:
```
2021-06-17T07:21:57.2207974Z ___________________ test_servicebus_message_repr_with_props ____________________
2021-06-17T07:21:57.2212410Z 
2021-06-17T07:21:57.2213432Z     def test_servicebus_message_repr_with_props():
2021-06-17T07:21:57.2214516Z         message = ServiceBusMessage(
2021-06-17T07:21:57.2215552Z             body="hello",
2021-06-17T07:21:57.2216947Z             application_properties={'prop': 'test'},
2021-06-17T07:21:57.2230072Z             session_id="id_session",
2021-06-17T07:21:57.2231277Z             message_id="id_message",
2021-06-17T07:21:57.2232473Z             scheduled_enqueue_time_utc=datetime.now(),
2021-06-17T07:21:57.2233686Z             time_to_live=timedelta(seconds=30),
2021-06-17T07:21:57.2234921Z             content_type="content type",
2021-06-17T07:21:57.2236123Z             correlation_id="correlation",
2021-06-17T07:21:57.2237283Z             subject="github",
2021-06-17T07:21:57.2238427Z             partition_key="id_session",
2021-06-17T07:21:57.2239597Z             to="forward to",
2021-06-17T07:21:57.2240660Z             reply_to="reply to",
2021-06-17T07:21:57.2241659Z >           reply_to_session_id="reply to session"
2021-06-17T07:21:57.2242616Z             )
2021-06-17T07:21:57.2243542Z 
2021-06-17T07:21:57.2245062Z azure-sdk-for-python/sdk/servicebus/azure-servicebus/tests/test_message.py:37: 
2021-06-17T07:21:57.2246547Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-06-17T07:21:57.2248760Z azure-sdk-for-python/sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py:123: in __init__
2021-06-17T07:21:57.2250075Z     self.time_to_live = kwargs.pop("time_to_live", None)
2021-06-17T07:21:57.2251285Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2021-06-17T07:21:57.2251923Z 
2021-06-17T07:21:57.2255473Z self = ServiceBusMessage(body=hello, application_properties={'prop': 'test'}, session_id=id_session, message_id=id_message, c...ssion, subject=github, time_to_live=0:00:30, partition_key=None, scheduled_enqueue_time_utc=2021-06-17 07:21:01.716564)
2021-06-17T07:21:57.2257417Z value = datetime.timedelta(seconds=30)
2021-06-17T07:21:57.2258079Z 
2021-06-17T07:21:57.2259079Z     @time_to_live.setter
2021-06-17T07:21:57.2260229Z     def time_to_live(self, value):
2021-06-17T07:21:57.2264735Z         # type: (datetime.timedelta) -> None
2021-06-17T07:21:57.2266007Z         if not self._raw_amqp_message.header:
2021-06-17T07:21:57.2269992Z             self._raw_amqp_message.header = AmqpMessageHeader()
2021-06-17T07:21:57.2272132Z         if value is None:
2021-06-17T07:21:57.2273272Z             self._raw_amqp_message.header.time_to_live = value
2021-06-17T07:21:57.2274503Z             if self._raw_amqp_message.properties.absolute_expiry_time:
2021-06-17T07:21:57.2275968Z                 self._raw_amqp_message.properties.absolute_expiry_time = value
2021-06-17T07:21:57.2277155Z         elif isinstance(value, datetime.timedelta):
2021-06-17T07:21:57.2278377Z             self._raw_amqp_message.header.time_to_live = value.seconds * 1000
2021-06-17T07:21:57.2279482Z         else:
2021-06-17T07:21:57.2280581Z             self._raw_amqp_message.header.time_to_live = int(value) * 1000
2021-06-17T07:21:57.2281857Z     
2021-06-17T07:21:57.2282873Z         if self._raw_amqp_message.header.time_to_live and \
2021-06-17T07:21:57.2284120Z                 self._raw_amqp_message.header.time_to_live != MAX_DURATION_VALUE:
2021-06-17T07:21:57.2285391Z             if not self._raw_amqp_message.properties:
2021-06-17T07:21:57.2286596Z                 self._raw_amqp_message.properties = AmqpMessageProperties()
2021-06-17T07:21:57.2288110Z >           self._raw_amqp_message.properties.creation_time = int(time.mktime(utc_now().timetuple()))
2021-06-17T07:21:57.2289371Z E           OverflowError: mktime argument out of range
2021-06-17T07:21:57.2289989Z 
2021-06-17T07:21:57.2291582Z azure-sdk-for-python/sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py:358: OverflowError
```

**Comment by yunhaoling** [comment]
on windows, the following piece would give `OverflowError: mktime argument out of range`
```python
import time
t = (1969, 3, 2, 0, 0, 0, 0, 0, 0)  # less than 1970 01 01
ser = time.mktime(t)
print(ser)
```

stackoverflow issue: https://stackoverflow.com/questions/2518706/python-mktime-overflow-error

**Comment by yunhaoling** [comment]
pr merged, it should be fixed in the coming 7.3.1 release.
